container_service_extension/utils.py

```diff
@@ -365,3 +367,19 @@ def wrapper(*args, **kwargs):
 def generate_thread_name(function_name):
     parent_thread_id = threading.current_thread().ident
     return function_name + ':' + str(parent_thread_id)
+
+
+def use_mqtt_protocol(config):
```

Comment:
Probably this is not directly related to : 
use_mqtt_protocol function;
but what will CSE do in case both AMQP/MQTT config sections are present in config.yaml?
Is there a code making sure such confusing configs are invalidated? Do we need to add such check somewhere?

Tagging Aritra @rocknes in case we need some more clarity/explanation here.

Reply:
As of now, if there is the presence of the mqtt section in config.yaml and the vcd api version is greater than or equal to the minimum required version for MQTT, then mqtt will be used even in the presence of an amqp section. Otherwise, amqp will be used. I agree we could logic to make sure only one of these fields is present would be helpful too. @rocknes What do you think?